src/main.c

```diff
@@ -657,38 +668,35 @@ start_standout(void)
 }
 
 static void
-print_at(int y, int x, char *string, int max_length)
+end_standout(void)
 {
-	int	i;
-
-	move_to(y, x);
-
-	for (i = 0; string[i] != '\0' && i < max_length; i++) {
-		if (tty_putc(string[i]) == EOF)
-			err(1, "tty_putc");
-	}
+	tty_putp(exit_standout_mode);
 }
 
 static void
-move_to(int y, int x)
+print_query(char *query, size_t length, int position)
 {
-	tty_putp(tgoto(cursor_address, x, y));
-}
+	int	i;
 
-static void
-end_standout(void)
-{
-	tty_putp(exit_standout_mode);
+	tty_putp(restore_cursor);
+	put_line(query, length, 0);
+
+	tty_putp(restore_cursor);
+	for (i = 0; i < position; ++i)
+		tty_putp(cursor_right);
 }
 
 static int
 print_choices(int selection)
 {
 	char		*line;
-	int		 i, visible_choices_count = 0;
+	int		 visible_choices_count = 0;
 	size_t		 length, line_length = 64;
 	struct choice	*choice;
 
+	/* Emit query line. */
+	tty_putc('\n');
```

Comment:
Probably but you then need to move the cursor to the right position on query
line after printing the matching choices.

On Wed, Dec 16, 2015 at 12:31:51AM -0800, Calle Erlandsson wrote:
> >  	size_t		 length, line_length = 64;
> >  	struct choice	*choice;
> >  
> > +	/* Emit query line. */
> > +	tty_putc('\n');
> 
> Could this be avoided by printing the query before printing the choices?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/thoughtbot/pick/pull/79/files#r47747714